tests/generators/test_generators.py

```diff
@@ -151,3 +152,47 @@ def test_generator_structure(classname):
         "prompt" in inspect.signature(g.generate).parameters
     ), f"{classname}.generate() must accept parameter prompt"
     # generate("") w/ empty string doesn't fail, does return list
+
+
+TESTABLE_GENERATORS = [
+    classname
+    for classname in GENERATORS
+    if classname
+    not in [
+        "generators.function.Multiple",  # requires mock local function not implemented here
+        "generators.function.Single",  # requires mock local function not implemented here
+        "generators.ggml.GgmlGenerator",  # validates files on disk tested in own test class
+        "generators.guardrails.NeMoGuardrails",  # requires nemoguardrails as thirdy party install dependency
```

Comment:
Interesting reason. Isn't this true for most generators?

Reply:
Oddly we have some cases where the generator requires dependencies that are not in `pyproject.toml`. Examples I found are `guardrails.NeMoGuardrails` and `huggingFace.OptimumPipeline` and `openai_v0.OpenAIGeneratorv0`.

Other generators have all had the dependencies listed in `.toml`, since I don't know the current reasoning for ones other than `openai_v0` it seemed best to just document what I find.

Reply:
Huh. Well, nobody's a fan of inconsistent policy. #709 will remove one of the inconsistent cases, and I guess the other two can be mopped up elsewhere. Thanks for flagging.